example/clippedEdges.js

```diff
@@ -418,11 +426,38 @@ function render() {
 				if ( localPlane.intersectLine( tempLine, tempVector ) ) {
 
 					posAttr.setXYZ( index, tempVector.x, tempVector.y, tempVector.z );
+					intersectPoints.push( {
+						x: tempVector.x, y: tempVector.y, z: tempVector.z
+					} );
 					count ++;
 					index ++;
 
 				}
 
+				// When the plane passes through a vertex and one of the edges of the triangle, there will be three intersections, two of which must be repeated
+				if ( count === 3 ) {
+
+					// If the last point is a duplicate intersection
+					if (
+						( intersectPoints[ count - 1 ].x === intersectPoints[ count - 2 ].x && intersectPoints[ count - 1 ].y === intersectPoints[ count - 2 ].y )
+							|| ( intersectPoints[ count - 1 ].x === intersectPoints[ count - 3 ].x && intersectPoints[ count - 1 ].y === intersectPoints[ count - 3 ].y )
```

Comment:
Sorry to be unclear. I think something like this would be most readable and avoid creating unnecessary memory in the loop:

```js
if ( count === 3 ) {

    tempVec0.fromBufferAttribute( posAttr, index - 3 );
    tempVec1.fromBufferAttribute( posAttr, index - 2 );
    tempVec2.fromBufferAttribute( posAttr, index - 1 );

    // If the last point is a duplicate intersection
    if ( tempVec2.equals( tempVec0 ) || tempVec2.equals( tempVec1 ) ) { // ...
```

Reply:
Thank you for your guidance. I don't know much about threej. I will try again in this way!